core/src/jwk.rs

```diff
@@ -65,6 +67,24 @@ impl Jwk {
         jwk.key_id.clone_from(&self.key_id);
         Ok(Self(jwk))
     }
+
+    /// Resolve a DID and return the DIDDocument
+    ///
+    /// The DID Document contains the public key. For more info, see:
+    /// https://www.w3.org/TR/did-core/#dfn-did-documents
+    pub async fn resolve_did(did: &str) -> Option<DidDocument> {
+        let resolver = &did_method_key::DIDKey;
+        let input_metadata = ResolutionInputMetadata::default();
+        let base_did = match did.split('#').next() {
```

Comment:
`let base = did.split_once("#").map_or(did, |(b, _| b)` is probably nicer/more correct. I'm not sure if it's allowed, but this would still work if it didn't include the #.

Reply:
Nice catch! This is a better implementation!